public/app/plugins/panel/traces/TagsEditor.tsx

```diff
@@ -0,0 +1,117 @@
+import { css } from '@emotion/css';
+import { useMemo } from 'react';
+
+import { GrafanaTheme2, StandardEditorProps, toOption } from '@grafana/data';
+import { IconButton, Field, useStyles2, Select } from '@grafana/ui';
+
+import { newTagFilter, Tag } from '../../../features/explore/TraceView/useSearch';
+import { getTraceTagKeys, getTraceTagValues } from '../../../features/explore/TraceView/utils/tags';
+import { transformDataFrames } from '../../../features/explore/TraceView/utils/transform';
+
+type Props = StandardEditorProps<Tag[], unknown, Tag[]>;
+
+export const TagsEditor = ({ value, onChange, context }: Props) => {
+  const styles = useStyles2(getStyles);
+
+  const trace = useMemo(() => transformDataFrames(context.data[0]), [context.data]);
+  const tagKeys = useMemo(() => (trace ? getTraceTagKeys(trace).map(toOption) : []), [trace]);
+  const operators = [toOption('='), toOption('!='), toOption('=~'), toOption('!~')];
+
+  if (!value || value.length < 1) {
+    value = [newTagFilter()];
+  }
+
+  const addTag = () => {
+    onChange([...value, newTagFilter()]);
+  };
+
+  const removeTag = (idx: number) => {
+    const copy = value.slice();
+    copy.splice(idx, 1);
+    onChange(copy);
+  };
+
+  const setKey = (idx: number, key?: string) => {
+    const copy = value.slice();
+    copy[idx].key = key;
+    copy[idx].value = undefined;
+    if (copy.length === 0) {
+      onChange(undefined);
+    } else {
+      onChange(copy);
+    }
+  };
+
+  const setOperator = (idx: number, operator?: string) => {
+    const copy = value.slice();
+    copy[idx].operator = operator ?? '=';
+    if (copy.length === 0) {
+      onChange(undefined);
+    } else {
+      onChange(copy);
+    }
+  };
+
+  const setValue = (idx: number, val?: string) => {
+    const copy = value.slice();
+    copy[idx].value = val;
+    if (copy.length === 0) {
+      onChange(undefined);
+    } else {
+      onChange(copy);
+    }
+  };
+
+  return (
+    <ul className={styles.list}>
+      {value.map((tag, idx) => (
+        <li key={`${idx}.${tag.id}`}>
+          <div className={styles.listItem}>
+            <Field label={'Key'} style={{ width: '40%' }}>
+              <Select
+                id={`tag-key-${tag.id}`}
+                onChange={(v) => setKey(idx, v.value)}
```

Comment:
I would also reset the value if the key changes for UX.

Is there a particular reason you didn't try to reuse the existing tags section on the search tab? We have lots of extra functionality and usability improvements in there.